charmhub/refresh.go

```diff
@@ -407,3 +426,27 @@ func composeMetadataHeaders(a, b Headers) Headers {
 	}
 	return result
 }
+
+// validatePlatform ensures that we do not pass "all" as part of platform.
+// This function is to help find programming related failures.
+func validatePlatform(rp RefreshPlatform) error {
```

Comment:
On the fence about leaving this validate code here.  Used it to find areas to fix.